tensorflow/tensorboard/components/vz_line_chart/vz-line-chart.ts

```diff
@@ -440,26 +440,18 @@ module VZ {
       let factor = (Math.pow(1000, this.smoothingWeight) - 1) / 999;
       let data = dataset.data();
       let kernelRadius = Math.floor(data.length * factor / 2);
-
-      data.forEach((d, i) => {
-        let actualKernelRadius = Math.min(kernelRadius, i);
-        let start = i - actualKernelRadius;
-        let end = i + actualKernelRadius + 1;
-        if (end >= data.length) {
-          // In the beginning, it's OK for the smoothing window to be small,
-          // but this is not desirable towards the end. Rather than shrinking
-          // the window, or extrapolating data to fill the gap, we're simply
-          // not going to display the smoothed line towards the end.
-          d.smoothed = Infinity;
-        } else if (!_.isFinite(d.scalar)) {
-          // Only smooth finite numbers.
+      data.forEach(function (d, i) {
+        var actualKernelRadius = Math.min(kernelRadius, i)
+        var start = Math.max(0, i - actualKernelRadius);
```

Comment:
Isn't the Math.max(0, i - actualKernelRadius) unnecessary?
Because when (i <= kernelRadius), actualKernelRadius = min(kernelRadius, i) = i
so i - actaulKernelRadius == 0 in that case.

conversely, if (i > kernelRadius), then (i - actualKernelRadius > 0)

so the Max(0) never gets invoked